envi/config.py

```diff
@@ -217,9 +217,15 @@ def saveConfigFile(self, filename=None):
         '''
         if filename is None:
             filename = self.filename
+        base = os.path.dirname(filename)
+        if not os.path.exists(base):
+            try:
+                os.makedirs(base)
+            except Exception as err:
+                logger.warning('FIXME - invalid homedir, playing along... (%s)', err)
 
         cfgdict = self.getConfigPrimitive()
-        with open(filename, 'wb') as fd:
+        with open(filename, 'w') as fd:
```

Comment:
this should almost certainly be `rb` or using the `encoding=` kwarg with `rt`.

Reply:
So this is actually a direct response to #350, and I can confirm the 'wb'/'rb' does make it throw when saving. And unless something has changed, I was under the impression that 't' was the default and has been for quite a while?

Reply:
i raise this because i've been bitten in the past due to the encoding being platform dependent:

![image](https://user-images.githubusercontent.com/156560/107697023-eab78f00-6c6f-11eb-9ca1-a38ab9f854da.png)

i.e. configs created on windows wont work on linux. given that the config doesn't contain platform-specific information, i think the config should be portable across platforms. so, would recommend `encoding=` (with explicit `mode=wt` to be clear) or using `wb` with explicit calls to `bytes.encode(...)`.

Reply:
Ahh that's a fair point that I tend to not run into much. I agree there isn't anything platform specific in the config, so it should be portable. 

I'll update it to be encoding='utf-8' and mode='wt'/'rt'.